### PR TITLE
Added user, role, policy for Jenkins user

### PR DESCRIPTION
**Pull request by bmcclellan-cu**
## Checklist
- [ ] Changes are fully implemented without dangling issues or TODO items
- [ ] Deprecated/superseded code is removed or marked with deprecation warning
- [ ] Current dependencies have been properly specified and old dependencies removed
- [ ] New code/functionality has accompanying tests and any old tests have been updated to match any new assumptions
- [ ] The changelog.md has been updated

### CODE REVIEW

File: lasp_opensearch_data_center/constructs/frontend.py [hunk 1]
```diff
@@ -121,25 +122,30 @@ def __init__(
             }
         )
 
-        # Define CI/CD group and policy names
-        frontend_group_name = f"{account_type}FrontEndGroup"
-        frontend_policy_name = f"{account_type}-frontend-automated-deploy"
+        # Define CI/CD user, role, and policy names
+        frontend_role_name = "jenkins-frontend-deploy-role"
```
Comment (medley56) [inline lasp_opensearch_data_center/constructs/frontend.py:126]: I suggest we leave all these names and IDs as generic as possible. e.g. "frontend-deploy-role"

File: lasp_opensearch_data_center/constructs/frontend.py [hunk 2]
```diff
@@ -164,6 +170,9 @@ def __init__(
             ],
         )
 
+        # Currently the /frontend object path in the S3 bucket is restricted to entities with this tag
+        Tags.of(self.frontend_deployment_role).add("group", "frontend")
```
Comment (medley56) [inline lasp_opensearch_data_center/constructs/frontend.py:174]: Can you just remind me why we did this based on tags? Obviously it works well but why not just directly reference the bucket ARN?

**Review by medley56 — COMMENTED** [review]
I'd like to remove "jenkins" from as many things as possible so future users can deploy however they want without confusing semantics.

File: lasp_opensearch_data_center/constructs/frontend.py (hunk 1 above)
Comment (bmcclellan-cu) [reply lasp_opensearch_data_center/constructs/frontend.py:126]: Sounds good, simple fix.

File: lasp_opensearch_data_center/constructs/frontend.py (hunk 2 above)
Comment (bmcclellan-cu) [reply lasp_opensearch_data_center/constructs/frontend.py:174]: This was a pattern we had used in SpaceWeather to restrict specific folders to specific groups.

I'll remove this tag, as we already have it locked down to the frontend/ object path in lines 163-167:

                    resources=[
                        # All bucket objects must start with the keyword "frontend"
                        # Additional restrictions imposed at the resource/bucket level
                        frontend_bucket.bucket_arn,
                        frontend_bucket.bucket_arn + "/frontend/*",


Any other updates you'd like here?

Comment (bmcclellan-cu) [reply lasp_opensearch_data_center/constructs/frontend.py:174]: As part of that I'd remove this entire block from the frontend_storage construct as well:

 # Restrict puts/deletes to the "frontend/" folder to a specific tag
        # An IAM user must be generated for the webteam via the AWS CLI
        # and tagged with group:frontend
        #
        # For the DEV account the Web team will put their built JS apps into this object path:
        # "frontend/libera/live/"
        # For the PROD account the Web team will put their built JS apps into this versioned object path:
        # "frontend/libera/VXX/"
        #
        # All webteam PUTs will be performed via Jenkins using CLI credentials
        self.frontend_bucket.add_to_resource_policy(
            aws_iam.PolicyStatement(
                effect=aws_iam.Effect.DENY,
                principals=[aws_iam.AnyPrincipal()],
                actions=[
                    "s3:DeleteObject",
                    "s3:PutObject",
                ],
                resources=[
                    self.frontend_bucket.bucket_arn,
                    self.frontend_bucket.arn_for_objects("frontend/*"),
                ],
                conditions={"StringNotLike": {"aws:PrincipalTag/group": "*frontend*"}},
            )
        ),

Comment (medley56) [reply lasp_opensearch_data_center/constructs/frontend.py:174]: Yeah, I think that's a more visible way to manage permissions. It's easy to see which bucket has this permission. Hard to see which bucket has a specific tag.

**Pull request merged**